chain/syncer_test.go

```diff
@@ -770,6 +754,16 @@ func TestLoadFork(t *testing.T) {
 	require.NoError(t, err)
 	requireHead(t, chainStore, forklink3)
 
+	// Put blocks in global IPLD blockstore
+	// TODO #2128 make this cleaner along with broad test cleanup.
```

Comment:
We can make this cleaner (but more opaque) by forcing the blockSource (fetcher) to be backed by the same repo/globalIPLD store.

Reply:
I think the fetcher is supposed to represent network, so we won't be exercising the right paths if the stuff to be fetched is already local.